open_spiel/games/kriegspiel.cc

```diff
@@ -223,7 +223,7 @@ class KriegspielObserver : public Observer {
                           Allocator *allocator) const {
     int board_size = board.BoardSize();
     WriteBinary(msg.illegal, prefix + "_illegal", allocator);
-    WriteScalar(msg.capture_type, 0, 2, prefix + "_capture_type", allocator);
+    WriteScalar(static_cast<int8_t>(msg.capture_type), 0, 2, prefix + "_capture_type", allocator);
```

Comment:
Can you please check that msg.capture_type is in the proper range at the start of this function?